tsl/src/continuous_aggs/utils.c

```diff
@@ -707,3 +708,237 @@ continuous_agg_migrate_to_time_bucket(PG_FUNCTION_ARGS)
+		}
+
+		if (bf->bucket_time_offset != NULL)
+		{
+			bucket_offset = DatumGetCString(
+				DirectFunctionCall1(interval_out, IntervalPGetDatum(bf->bucket_time_offset)));
+		}
+	}
+	else
+	{
+		/* Bucketing on integers */
+		bucket_width = palloc0(MAXINT8LEN + 1 * sizeof(char));
+		pg_lltoa(bf->bucket_integer_width, bucket_width);
+
+		/* Integer buckets with origin are not supported, so noting to do. */
+		Assert(bucket_origin == NULL);
+
+		if (bf->bucket_integer_offset != 0)
+		{
+			bucket_offset = palloc0(MAXINT8LEN + 1 * sizeof(char));
+			pg_lltoa(bf->bucket_integer_offset, bucket_offset);
+		}
+	}
+
+	tupdesc = BlessTupleDesc(tupdesc);
+
+	/* Bucket function */
+	ts_datum_set_objectid(Anum_cagg_bucket_function_oid, datums, bf->bucket_function);
+
+	/* Bucket width */
+	ts_datum_set_text_from_cstring(Anum_cagg_bucket_function_width, datums, bucket_width);
+
+	/* Bucket origin */
+	ts_datum_set_text_from_cstring(Anum_cagg_bucket_function_origin, datums, bucket_origin);
+
+	/* Bucket offset */
+	ts_datum_set_text_from_cstring(Anum_cagg_bucket_function_offset, datums, bucket_offset);
+
+	/* Bucket timezone */
+	ts_datum_set_text_from_cstring(Anum_cagg_bucket_function_timezone,
+								   datums,
+								   bf->bucket_time_timezone);
+
+	/* Bucket fixed width */
+	ts_datum_set_bool(Anum_cagg_bucket_function_fixed_width, datums, bf->bucket_fixed_interval);
+
+	Assert(tupdesc->natts == Natts_cagg_validate_query);
+	tuple = ts_heap_form_tuple(tupdesc, datums);
+
+	return HeapTupleGetDatum(tuple);
+}
+
+/*
+ * Get the bucket function information for the given materialized hypertable id.
+ *
+ * When running `cagg_get_bucket_function_info` the function returns the following fields:
+ * - oid: The Oid of the bucket function
+ * - width: The width of the bucket function
+ * - origin: The origin of the bucket function
+ * - offset: The offset of the bucket function
+ * - timezone: The timezone of the bucket function
+ * - fixed_width: Is the bucket width fixed
+ *
+ * When running `cagg_get_bucket_function` the function returns the following fields:
+ * - oid: The Oid of the bucket function
+ */
+static Datum
+cagg_get_bucket_function_datum(int32 mat_hypertable_id, FunctionCallInfo fcinfo)
+{
+	Oid direct_view_oid = get_direct_view_oid(mat_hypertable_id);
+	TupleDesc tupdesc;
+
+	if (fcinfo != NULL && get_call_result_type(fcinfo, NULL, &tupdesc) != TYPEFUNC_COMPOSITE)
+		elog(ERROR, "function returning record called in context that cannot accept type record");
+
+	ContinuousAggsBucketFunction *bf = ts_cagg_get_bucket_function_info(direct_view_oid);
+
+	if (!OidIsValid(bf->bucket_function))
+	{
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("time_bucket function not found in CAgg definition for "
+						"mat_ht_id: %d",
+						mat_hypertable_id)));
+		pg_unreachable();
+	}
+
+	if (fcinfo != NULL)
+		return create_cagg_get_bucket_function_datum(tupdesc, bf);
+
+	return ObjectIdGetDatum(bf->bucket_function);
```

Comment:
This do not seem to have coverage. Can you add some?

Reply:
Done!